autode/opt/coordinates/internals.py

```diff
@@ -47,10 +47,11 @@ def __new__(cls, input_array) -> "InternalCoordinates":
 
         arr = super().__new__(cls, input_array, units="Å")
 
-        arr._x = None
-        arr.primitives = None
+        arr._x = None  # Cartesian coordinates
+        arr._q = None  # Primitive values
+        arr.primitives = None  # Primitive coordinate set
```

Comment:
can we have typing rather than comments

Reply:
@t-young31 I tried adding type hints like this: `arr._x: Optional["CartesianCoordinates"] = None" but got the mypy error: "Type cannot be declared in assignment to non-self attribute". I think mypy only allows type hint in `__init__` but not `__new__`.

Reply:
ugh, annoying – thanks for the attempt